internal/pkg/policy/policy_output.go

```diff
@@ -246,6 +272,81 @@ func (p *Output) prepareElasticsearch(
 	return nil
 }
 
+func (p *Output) createAndGetBulker(ctx context.Context, mainBulker bulk.Bulk, outputMap map[string]map[string]interface{}) (bulk.Bulk, error) {
+	bulker := bulkerMap[p.Name]
+	if bulker != nil {
+		mainBulker.CheckRemoteOutputChanged(p.Name, outputMap[p.Name])
+		return bulker, nil
+	}
+	bulkCtx, bulkCancel := context.WithCancel(context.Background())
+	defer bulkCancel()
+	es, err := p.createRemoteEsClient(bulkCtx, outputMap)
+	if err != nil {
+		return nil, err
+	}
+	// starting a new bulker to create/update API keys for remote ES output
+	newBulker := bulk.NewBulker(es, mainBulker.Tracer())
+	bulkerMap[p.Name] = newBulker
+
+	errCh := make(chan error)
+	go func() {
+		runFunc := func() (err error) {
+			return newBulker.Run(bulkCtx)
+		}
+
+		errCh <- runFunc()
+	}()
+	go func() (err error) {
+		select {
+		case err = <-errCh:
+		case <-bulkCtx.Done():
+			err = bulkCtx.Err()
+		}
+		return
+	}()
+
+	return newBulker, nil
+}
+
+func (p *Output) createRemoteEsClient(ctx context.Context, outputMap map[string]map[string]interface{}) (*elasticsearch.Client, error) {
+	hostsObj := outputMap[p.Name]["hosts"]
+	hosts, ok := hostsObj.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to get hosts from output: %v", hostsObj)
+	}
+	hostsStrings := make([]string, len(hosts))
+	for i, host := range hosts {
+		hostsStrings[i], ok = host.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to get hosts from output: %v", host)
+		}
+	}
+
+	cfg := config.Config{
+		Output: config.Output{
+			Elasticsearch: config.Elasticsearch{
+				Hosts:        hostsStrings,
+				ServiceToken: p.ServiceToken,
```

Comment:
I'm wondering where the Advanced YAML config of an output should be used. How does it work for local ES output?

This is the `config_yaml` field in output, and appears in agent policy under output too.
<img width="656" alt="image" src="https://github.com/elastic/fleet-server/assets/90178898/b75b3ca8-bb78-4106-9728-103279480828">

<img width="668" alt="image" src="https://github.com/elastic/fleet-server/assets/90178898/11103268-dfd9-4e7c-ad3a-3291fb3180a2">

Reply:
The Advanced YAML box for an output is meant to allow adding configuration for that output that aren't exposed in the UI, worker and bulk_max_size are the two most common. All of the keys in the Advanced YAML box should get merged with the rest of the output configuration and sent to the agent. 

I don't see a reason why this would be different for a remote ES output.

Reply:
Okay, so I think it doesn't need any new logic, as the advanced config is already sent to the agent under outputs section in agent policy.